sigmastate/src/main/scala/sigmastate/UncheckedTree.scala

```diff
@@ -82,4 +82,14 @@ case class CThresholdUncheckedNode(override val challenge: Challenge,
   require(k >= 0 && k <= children.length)
 
   override val conjectureType = ConjectureType.ThresholdConjecture
+
+  override def canEqual(other: Any) = other.isInstanceOf[CThresholdUncheckedNode]
+  override def equals(other: Any) = (this eq other.asInstanceOf[AnyRef]) || (other match {
```

Comment:
Why ?

Reply:
Becase Challenge is tagged Array[Byte] and Array's equality is not compatible with case classes. This fix is to avoid subtle bugs.